wintermotes/week2/Ass3/script.js

```diff
@@ -0,0 +1,386 @@
+var MessageBox = React.createClass({
+
+			//Question 1: Is there no way i can pass a function to a component without having it to traverse the whole tree? 
+			//Question 2: what is the cost of passing down an array as "lists" down the tree? For instance, in setState i copy the state, and resets it for each render. 
+			// 			  isn't that expensive? 
+			//Question 3: How to create seperate files with different react classes, e.g a picture file with relevant react classes, a user file with relevant react classes etc. 
+			// 			  Code seems to be able to pile up pretty quickly 
+			//Question 4+5: Are listed in the code, since pure code is better than trying to explain the issue. 
+
+	    	getInitialState : function() {
+				var lists = [
+					{
+						listId : 0, 
+						listName : 'List1', 
+						messages : [
+							{messageId : 0, messageText : "List1 msg1", archived : false, listId : 0}, 
+							{messageId : 1, messageText : "List1 msg2", archived : false, listId : 0}
+						], 
+						archivedMessages : [
+							{messageId : 0, messageText : "Archived: List1 msg1", archived : true, listId : 0},
+							{messageId : 1, messageText : "Archived: List1 msg2", archived : true, listId : 0}
+						],
+					},
+					{
+						listId : 1, 
+						listName : 'List2', 
+						messages : [
+							{messageId : 0, messageText : "List2 msg1", archived : false, listId : 1}, 
+							{messageId : 1, messageText : "List2 msg2", archived : false, listId : 1},
+							{messageId : 2, messageText : "List2 msg3", archived : false, listId : 1}
+						], 
+						archivedMessages : [
+							{messageId : 0, messageText : "List1 msg1", archived : true, listId : 0},
+							{messageId : 1, messageText : "List1 msg2", archived : true, listId : 0}
+						],
+					}
+				];
+				return {lists};
+			},
+			createList: function(listName){
+				var listId = this.state.lists.length
+				var listObject = {listId : listId, listName : listName, messages : [], archivedMessages : []}
+				var newList = this.state.lists.concat(listObject)
+				
+				this.setState({lists: newList})
+			},
+			createMessage: function(messageObject, listId) {
+				var newLists = this.state.lists
+				if(messageObject.archived == false)
+					newLists[listId].messages.push(messageObject)
+				else 
+					newLists[listId].archivedMessages.push(messageObject)
+
+				this.setState({lists : newLists})
+			},
+			moveMessage: function(listId, newListId, messageId){
+				var lists = this.state.lists
+				var messageToMove = lists[listId].messages[messageId]
```

Comment:
`messageToMove` still points to the state. Please deep copy the message by using `var messageToMove = Object.assign({}, lists[listId].messages[messageId])`